cpp/src/arrow/ArrowConfig.cmake.in

```diff
@@ -38,45 +38,42 @@ include("${CMAKE_CURRENT_LIST_DIR}/ArrowOptions.cmake")
 
 include(CMakeFindDependencyMacro)
 
-if(ARROW_USE_STATIC)
-  set(LIB_PATH_SUFFIXES
-      "${CMAKE_LIBRARY_ARCHITECTURE}"
-      "lib/${CMAKE_LIBRARY_ARCHITECTURE}"
-      "lib64"
-      "lib32"
-      "lib"
-      "bin"
-      "Library"
-      "Library/lib"
-      "Library/bin")
-  set(INCLUDE_PATH_SUFFIXES "include" "Library" "Library/include")
-
-  set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
-  set(THREADS_PREFER_PTHREAD_FLAG TRUE)
-  find_dependency(Threads)
-
-  if(DEFINED CMAKE_MODULE_PATH)
-    set(_CMAKE_MODULE_PATH_OLD ${CMAKE_MODULE_PATH})
-  endif()
-  set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
-
-@ARROW_FIND_DEPENDENCY_LIST@
-
-  if(DEFINED _CMAKE_MODULE_PATH_OLD)
-    set(CMAKE_MODULE_PATH ${_CMAKE_MODULE_PATH_OLD})
-    unset(_CMAKE_MODULE_PATH_OLD)
-  else()
-    unset(CMAKE_MODULE_PATH)
-  endif()
-endif()
-
-
 # Load targets only once. If we load targets multiple times, CMake reports
 # already existent target error.
 if(NOT (TARGET arrow_shared OR TARGET arrow_static))
   include("${CMAKE_CURRENT_LIST_DIR}/ArrowTargets.cmake")
 
   if(TARGET arrow_static)
+    set(LIB_PATH_SUFFIXES
```

Comment:
This change means that the static dependencies need to be resolved even if `arrow_static` is never used.

Reply:
Yes. Is it strange?
Do you know a CMake convention when a CMake package provides both shard and static libraries?
Do users need to specify "we want to use static library" explicitly?

Should we use the `COMPONENTS` feature in `find_package`?